### DISCUSSION TITLE
Timezone library for converting timezone

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#11848] [Ideas]
I'm now converting timezone with daylight-saving by libc functions like tzset in Linux, but need v library for cross-platform and multi-thread.

**Comment by ghost** [comment]
https://github.com/derickr/timelib

**Comment by dumblob** [comment]
Just some thoughts.

How do you want to manage the lifetime of the timezone definitions? I mean, they're slightly changing all the time based on political situation and I'd like *V* to be realiable even for e.g. hydronautics, aeronautics, etc. Maybe include versioning of the TZ data directly into the datum structure **and** API (and returning `error( ... )` in case there is a version mismatch)? This mismatch really happens not just in hydronautics/aeronautics, but also in military (and that's plenty of SW!) - they just decide on their own which TZ (and DST) definitions for which projects and for which cases they'll use and thus there are always boundary interfaces between different projects/cases. Baking this directly into the data representation and thus into the API makes the life so much easier and makes all these issues explicit.

**Comment by ghost** [comment]
#11644
